packages/discovery-react-components/src/components/DocumentPreview/utils/documentData.ts

```diff
@@ -49,24 +50,27 @@ export function detectPreviewType(
 ): PreviewType {
   const fileType = document.extracted_metadata?.file_type;
   // passages contain location offsets against text-based strings (not HTML)
-  const hasPassage = highlight && 'passage_text' in highlight;
   const hasTextMappings = !!document.extracted_metadata?.text_mappings;
+  const hasHighlight = !!highlight;
+  const isHighlightPassage = isPassage(highlight);
+  const isHighlightTable = isTable(highlight);
 
   if (fileType === 'pdf' && file) {
-    // If there is a passage to highlight, text_mappings are required to map
+    // When trying to highlight a passage or table, text_mappings are required to map
     // between passages' text-based offsets and the BBOX data need to highlight
     // on PDFs
-    if (hasTextMappings || !hasPassage) {
+    if (!hasHighlight || hasTextMappings) {
       return 'PDF';
     }
   }
 
   const isJsonType = isJsonFile(document);
   const isCsvType = isCsvFile(document);
   if (document.html && !isJsonType && !isCsvType) {
-    // HTML view cannot display a passage highlight unless the document have text_mappings.
-    // So, do not show as HTML when the document have a passage but does not have text_mappings.
-    if (hasTextMappings || !hasPassage) {
+    // When trying to highlight a passage, only show as HTML when the document has text_mappings.
+    // (since HTML view cannot display a passage highlight unless the document have text_mappings)
+    // When trying to highlight a table, text mappings aren't needed, so display HTML
+    if (!hasHighlight || (isHighlightPassage && hasTextMappings) || isHighlightTable) {
```

Comment:
Maybe I'm confused, but won't the case where `!hasHighlight` always get swept up by the block of logic returning `PDF` at lines 62-64 (at time of writing). Meaning that case would never get caught here to return `HTML`.

Reply:
Both of these if statements are nested within other ifs. For clarity, here's the code without the comments and git +/-.

```
if (fileType === 'pdf' && file) {
    if (!hasHighlight || hasTextMappings) {
      return 'PDF';
    }
}

if (document.html && !isJsonType && !isCsvType) {
  if (!hasHighlight || (isHighlightPassage && hasTextMappings) || isHighlightTable) {
    return 'HTML';
  }
}
```

So, a non-PDF, non-JSON, non-CSV file without a highlight and with html data will end up returning `'HTML'`. I considered condensing them to single if statements, but I thought that would be less clear (i.e. too many variables in one if).